docker/benchbase/Dockerfile

```diff
@@ -0,0 +1,81 @@
+# A simple dev environment container for benchbase.
+#
+# - Build:
+#
+#   - Full image:
+#
+#     docker build -t benchbase -f Dockerfile ../..
+#
+#   - Dev only:
+#
+#     # Skip copying and building the source into the devcontainer image since we will map it in later.
+#     docker build -t benchbase-dev -f Dockerfile --target devcontainer ../..
+#
+# - Run:
+#
+#   - Full image:
+#
+#     # Map the config to read in and a place for the results to be written out.
+#     docker run -it --rm --name benchbase -v $PWD/config:/benchbase/config -v $PWD/results:/benchbase/results --env BENCHBASE_PROFILE=postgres benchbase -- <benchbase args>
+#
+#   - Dev image:
+#
+#     # Map the whole source directory into the container.
+#     # Optionally build the source as the container launch executable.
+#     docker run -it --rm --name benchbase-dev -v $PWD:/benchbase benchbase-dev mvn clean package -P postgres
+
+FROM --platform=linux maven:3.8.5-eclipse-temurin-17 AS devcontainer
+
+# Add a containeruser that allows vscode/codespaces to map the local host user
+# (often uid 1000) to some non-root user inside the container.
+ARG UID=1000
+ARG GID=1000
+RUN addgroup --gid ${GID} containergroup \
+    && adduser --disabled-password --gecos 'Container User' --uid ${UID} --gid ${GID} containeruser
+RUN mkdir -p /benchbase/results && chown -R containeruser:containergroup /benchbase/
+USER containeruser
+ENV MAVEN_CONFIG=/home/containeruser/.m2
+WORKDIR /benchbase
+
+# When running the devcontainer, just launch an interactive shell by default.
+ENTRYPOINT ["/bin/bash", "-l"]
+
+# Copy the full source into the container image.
+# Assumes the context is given as the root of the repo.
+
+# Preload some dependencies.
+COPY --chown=containeruser:containergroup pom.xml /benchbase/pom.xml
+COPY --chown=containeruser:containergroup .git/ /benchbase/.git/
+ARG MAVEN_OPTS
+RUN mvn -B --file pom.xml initialize \
+    && rm -rf /benchbase/.git /benchbase/target /benchbase/pom.xml
+
+# Add an additional layer that also includes a built copy of the source.
+FROM devcontainer AS fullimage
```

Comment:
FWIW, this is useful for deploying the image as benchmarking agent in k8s, Azure Batch, etc.
However, it probably doesn't need a full JDK at that point.
One can save a little bit of space by removing the `/benchbase/src` tree and the `openssh-client` (and `apt autoremove`ing to get rid of the dependencies), but the real wins are in having the fullimage use the JRE instead of the JDK.
I'm gonna call that future improvements for now though.